src/string.cr

```diff
@@ -489,13 +519,10 @@ class String
   end
 
   # :nodoc:
-  record ToU64Info,
-    value : UInt64,
-    negative : Bool,
-    invalid : Bool
+  record UIntInfo, value : (UInt64 | UInt128), negative : Bool, invalid : Bool
```

Comment:
It's better to keep a `ToU64Info` and `ToU128Info` so parsing does not always involve a 128 bits integer as a representation and 64 bits are used when possible.

Or a generic can be used if preferred, but I think two structs will be good enough.